File: src/mturk/datatypes/assignment.clj
(ns mturk.assignment
  (:require [date-time])
  (:import [java.util Date])
  (:gen-class)
)

(def ^{:private true} assignmentStatusList '(:Submitted :Approved :Rejected))

(defreford Assignment [^String AssignmentId
                       ^String WorkerId
                       ^String HITId
                       ^Keyword AssignmentStatus
                       ^Date AutoApprovalTime 
                       ^Date AcceptTime
                       ^Date SubmitTime
                       ^Date ApprovalTime
                       ^Date RejectionTime 
                       ^Date Deadline
                       ^String Answer
                       ^String RequesterFeedback])


  

File: src/main/java/mturk/Assignment.java
package mturk;

//import mturk.core;
import java.util.Date;

public class Assignment {

    public enum AssignmentStatus { Submitted, Approved, Rejected; };
    
    private String assignmentId;
    private String workerId;
    private String HITId;
    private AssignmentStatus assignmentStatus;
    private Date autoApprovalTime;
    private Date acceptTime;
    private Date submitTime;
    private Date approvalTime;
    private Date rejectionTime;
    private Date deadline;
    private String answer;
    private String requesterFeedback;

    protected Assignment(String assignmentId,
		       String workerId,
		       String HITId,
		       AssignmentStatus assignmentStatus,
		       Date autoApprovalTime,
		       Date acceptTime,
		       Date submitTime,
		       Date approvalTime,
		       Date rejectionTime,
		       Date deadline,
		       String answer,
		       String requesterFeedback){
	this.assignmentId = assignmentId;
	this.workerId = workerId;
	this.HITId = HITId;
	this.assignmentStatus = assignmentStatus;
	this.autoApprovalTime = autoApprovalTime;
	this.acceptTime = acceptTime;
	this.submitTime = submitTime;
	this.approvalTime = approvalTime;
	this.rejectionTime = rejectionTime;
	this.deadline = deadline;
	this.answer = answer;
	this.requesterFeedback = requesterFeedback;
    }


    public static Assignment getAssignment(String assignmentId){
    	return null;
    }
    
}
